onebusaway-android/src/main/java/org/onebusaway/android/tripservice/PollerTask.java

```diff
@@ -120,55 +121,51 @@ private void poll1(Cursor c) {
 
         final String tripId = c.getString(COL_TRIP_ID);
         final String stopId = c.getString(COL_STOP_ID);
-        final long reminderMS = getReminderMS(tripId, stopId);
+        final long reminderMin = getReminderMin(tripId, stopId);
 
         ObaArrivalInfoResponse response = ObaArrivalInfoRequest
                 .newRequest(mContext, stopId).call();
 
-        Long departMS = null;
+        // Arrival information
+        ArrivalInfo arrivalInfo = null;
         if (response.getCode() == ObaApi.OBA_OK) {
-            departMS = checkArrivals(response, c);
+            arrivalInfo = checkArrivals(response, c);
         }
 
-        if (departMS != null) {
-            final long diffTime = departMS - System.currentTimeMillis();
-            if (diffTime <= reminderMS) {
+        if (arrivalInfo != null) {
+            if (arrivalInfo.getEta() <= reminderMin) {
```

Comment:
We're losing precision in our calculation here by changing from milliseconds to minutes, but I think it's ok in this context - our drop down to choose reminder times is rounded to the minute, and the values stored in the database for the reminder is also rounded to minutes.  I'd rather be consistent in calculating ETA throughout the application, and we can tackle application-wide precision increases (e.g., https://github.com/OneBusAway/onebusaway-android/issues/498) separately.